lxd/storage/drivers/driver_powerflex.go

```diff
@@ -0,0 +1,208 @@
+	}
+
+	// Detect and record the version.
+	out, err := shared.RunCommand("nvme", "version")
+	if err != nil {
+		return err
+	}
+
+	fields := strings.Split(strings.TrimSpace(out), " ")
+	if strings.HasPrefix(out, "nvme version ") && len(fields) > 2 {
+		powerFlexVersion = fmt.Sprintf("%s (nvme-cli) / x.y.z (sdc)", fields[2])
+	}
+
+	powerFlexLoaded = true
+	return nil
+}
+
+// isRemote returns true indicating this driver uses remote storage.
+func (d *powerflex) isRemote() bool {
+	return true
+}
+
+// Info returns info about the driver and its environment.
+func (d *powerflex) Info() Info {
+	return Info{
+		Name:              "powerflex",
+		Version:           powerFlexVersion,
+		OptimizedImages:   true,
+		PreservesInodes:   false,
+		Remote:            d.isRemote(),
+		VolumeTypes:       []VolumeType{VolumeTypeCustom, VolumeTypeVM, VolumeTypeContainer, VolumeTypeImage},
+		BlockBacking:      true,
+		RunningCopyFreeze: true,
+		DirectIO:          true,
+		IOUring:           true,
+		MountedRoot:       false,
+	}
+}
+
+// FillConfig populates the storage pool's configuration file with the default values.
+func (d *powerflex) FillConfig() error {
+	if d.config["powerflex.user.name"] == "" {
+		d.config["powerflex.user.name"] = powerFlexDefaultUser
+	}
+
+	if d.config["powerflex.mode"] == "" {
+		if d.canSDC() {
+			d.config["powerflex.mode"] = "sdc"
+		}
+
+		// Prefer the NVMe/TCP mode.
+		if d.canNVMe() {
+			d.config["powerflex.mode"] = "nvme"
+		}
+	}
+
+	// Fail if no PowerFlex mode could be discovered.
+	if d.config["powerflex.mode"] == "" {
+		return fmt.Errorf("Failed to discover PowerFlex mode")
+	}
+
+	// PowerFlex volumes have to be at least 8GiB in size.
+	if d.config["volume.size"] == "" {
+		d.config["volume.size"] = powerFlexDefaultSize
+	}
+
+	return nil
+}
+
+// Create is called during pool creation and is effectively using an empty driver struct.
+// WARNING: The Create() function cannot rely on any of the struct attributes being set.
+func (d *powerflex) Create() error {
+	err := d.FillConfig()
+	if err != nil {
+		return err
+	}
+
+	// Discover one of the storage pools SDS services.
+	if d.config["powerflex.mode"] == "nvme" {
+		if d.config["powerflex.sds"] == "" {
+			pool, err := d.powerFlex().getStoragePool(d.config["powerflex.pool"])
+			if err != nil {
+				return err
+			}
+
+			// TODO: check status of SDS or if they are applicable (ipList.role, sdsState, membershipState, mdmConnectionState)
+			sds, err := d.powerFlex().getProtectionDomainSdsRelations(pool.ProtectionDomainID)
+			if err != nil {
+				return err
+			}
+
+			if len(sds) == 0 {
+				return fmt.Errorf("Failed to retrieve at least one SDS for the given storage pool: %q", pool.ID)
+			}
+
+			if len(sds[0].IPList) == 0 {
+				return fmt.Errorf("Failed to retrieve IP from SDS: %q", sds[0].Name)
+			}
+
+			d.config["powerflex.sds"] = sds[0].IPList[0].IP
+		}
+	}
+
+	if d.config["powerflex.mode"] == "sdc" {
+		return fmt.Errorf("not supported")
```

Comment:
Missing capital N

Reply:
Actually we should use ErrNotSupported here

Reply:
Removed the condition completely.